release.sh

```diff
@@ -192,11 +192,10 @@ function checkout_code {
 
 function update_version_to_release {
     cd $SOURCE_DIR
-    # Update Python _version.py
-    sed -i .bak "s@^__version__.*@__version__ = '$RELEASE_VERSION'@g" enterprise_gateway/_version.py
 
-    # Update Makefile
-    sed -i .bak "s@$CURRENT_VERSION@$RELEASE_VERSION@g" Makefile
+    # Update tbump-managed versions
+    pip install tbump
```

Comment:
That makes sense to me

Reply:
FWIW, I applied these changes in [this commit](https://github.com/jupyter-server/enterprise_gateway/commit/11ecf18e39c62b5fb3b8fa2bcd14dcca13b7fade).  I can merge onto this PR or add in later but would appreciate your review in either case.

Also, should the `pyproject.toml`'s `[build-system][requires]` list be included in `list-build-deps.py` so that `hatchling` gets installed?  Seems like it should.

Reply:
Would you happen to know what this is about? https://github.com/kevin-bates/enterprise_gateway/runs/6480277396?check_suite_focus=true#step:12:40

Reply:
Actually, maybe a better option is to have `dev` and `test` extras, and use `pip install -e "[dev,test]"` in the docs and the Makefile?  We shouldn't need `hatchling` outside of when it gets invoked by `build` or `pip`, so I don't think we need it in `dev` dependencies.

Tbump doesn't like you to have any uncommitted changes.  What is changing those two files during the run?

Reply:
> Actually, maybe a better option is to have dev and test extras, and use pip install -e "[dev,test]" in the docs and the Makefile? 

I'm good with that. Just not aware of the options.  Is this something you were going to do, or should I do this (although still a little unsure of pyproject.toml section names, etc.)?

> Tbump doesn't like you to have any uncommitted changes. What is changing those two files during the run?

I don't think `pre-commit` was run, so I suspect that's what's going on.  There's nothing I've introduced to explicitly change these files.

Tell you what.  Since the build is green with _this_ PR, I'm inclined to merge this PR, then follow up with some helper targets, etc. - unless you'd like to make those changes here.  At his point today I'm going to wait until tomorrow for the merge anyway, so we can talk further tomorrow if necessary.  Have a good evening!

Reply:
I'll probably do a sweep of all repos to do the "pip install -e ".[dev,test]"` thingy.

> Have a good evening!

You too!

Reply:
ok - sounds good.  Let's proceed with this PR as-is and we can circle back to add a helper target using the editable install later.